scripts/ci/check_compliance.py

```diff
@@ -679,6 +682,11 @@ def check_no_undef_outside_kconfig(self, kconf):
         "MYFEATURE",
         "MY_DRIVER_0",
         "NORMAL_SLEEP",  # #defined by RV32M1 in ext/
+        "NO_CONFIG_BLOBS",      # Used in modules/hostap
+        "NO_CONFIG_WRITE",      # Used in modules/hostap
+        "NO_PBKDF2",            # Used in modules/hostap
+        "NO_RANDOM_POOL",       # Used in modules/hostap
+        "NO_WPA",               # Used in modules/hostap
```

Comment:
> The upstream hostap project uses various CONFIG_* options
that we need to whitelist.

This is only a problem because you directly set those as compile definitions here:
https://github.com/zephyrproject-rtos/zephyr/blob/9062595a5be321c3b3c0c7ab05813253a3dc17d1/modules/hostap/CMakeLists.txt#L39

If instead the hostap Kconfig file defines equivalent Kconfig options as mentioned in other comment, then those changes / whitelisting wouldn't be needed.

Reply:
There are way too many configuration options provided by WPA suppplicant, so, IMHO, we shouldn't expose all of them as Kconfig options, instead abstract them with higher level defines.

Reply:
It would be good to leave the module design flexible so that Kconfig can be introduced on an as-needed basis. Likely the most common defines for the Zephyr platform can be selected from the start, and bring out only the most relevant / important ones initially. Not a huge worry from my perspective.

Reply:
> There are way too many configuration options provided by WPA suppplicant, so, IMHO, we shouldn't expose all of them as Kconfig options, instead abstract them with higher level defines.

Unsure what you mean with expose.
You can introduce them as promptless, meaning they will have the setting define, possibly with a help text describing it's functionality.
Being promptless means the value is not directly user configurable.

Minimal example without help text could be:
```
config NO_WPA
        bool
        default y
```

A big reason for having the hostap configs mapped into Kconfig is to expose to others that this Kconfig setting name is already in use.

Somewhere else in our Zephyr Kconfig code base someone could introduce a:
```
config CRYPTO_INTERNAL
    bool "FOO internal crypto implementation"
    default n
    help
      FOO internal crypto implementation having nothing in common with hostap, 
      but having name collision because it was not visible that same name is used by hostap. 
```

or someone may introduce a new Kconfig in Zephyr which are also used in hostap, but not exposed as such.

Hostap tests for `CONFIG_ECC`, `CONFIG_ECC`, `CONFIG_SHA384`, and similar.
Luckily Zephyr uses more specific names, such as `BT_MESH`, `BT_ECC`, and so.
But imagine Zephyr having a Kconfig being set to true which collides with a config that is used by hostap.

I did notice some of those are used in files we include in the Zephyr library.

Reply:
> Unsure what you mean with expose.

They don't have a corresponding Zephyr Kconfig.

> A big reason for having the hostap configs mapped into Kconfig is to expose to others that this Kconfig setting name is already in use.

Understood.

> I did notice some of those are used in files we include in the Zephyr library.

Oh, haven't realized that.